drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -82,7 +82,11 @@ def service_factory(create: true) # rubocop:disable Lint/UnusedMethodArgument
   # Type Factories
   def enrollment_factory(create: true) # rubocop:disable Lint/UnusedMethodArgument
     # The enrollment has already been created, so we can just return it
-    @enrollment_factory ||= owner.enrollment if owner.is_a?(Hmis::Hud::CustomAssessment)
+    @enrollment_factory ||= if owner.is_a?(Hmis::Hud::CustomAssessment)
+      owner.enrollment
+    elsif owner.is_a?(Hmis::Hud::Enrollment)
+      owner
+    end
```

Comment:
Is owner.enrollment ever expensive? Maybe this doesn't need memoization? Also maybe a case statement would be  a little cleaner
```suggestion
   case owner
   when Hmis::Hud::CustomAssessment
     owner.enrollment
   when Hmis::Hud::Enrollment
     owner
   end
```